moon-store-mac-os/moon-store-mac-os/Domain/Repositories/ProductRepository.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  ProductRepository.swift
+//  moon-store-mac-os
+//
+//  Created by Jose Luna on 12/23/24.
+//
+
+import Foundation
+
+final class ProductRepository: BaseNetworkService {
+    private let decoder: JSONDecoder
+    
+    init(decoder: JSONDecoder = .init()) {
+        self.decoder = decoder
+    }
+    
+    func getProducts() async throws -> [ProductModel] {
+        do {
+            let result = try await getData(for: MSEndpoint.products.path)
+            let productResponse = try decoder.decode(ProductResponse.self, from: result)
+            
+            if productResponse.data.isEmpty { throw MSError.noData }
```

Comment:
3 Cosas.
1: no legible
2: deberia de llamarse solo response, si esta dentro de la funcion product se intulle que es el response de product.
3: Si no hay data deberia de mandar un arreglo vacio, no un error, ya que no es un error en realidad. Se necesita manejar diferente un data vacia a un error
```suggestion
            if productResponse.data.isEmpty { 
                throw MSError.noData 
            }
```